octree/include/pcl/octree/octree_iterator.h

```diff
@@ -524,6 +523,77 @@ namespace pcl
       };
 
     //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    /** \brief @b Octree iterator class
+     * \note This class implements a forward iterator for traversing octrees in a breadth-first manner.
+     *       It only returns the node at a given depth in the octree, if any.
+     * \ingroup octree
```

Comment:
Can you switch the description to "Iterator over all existing nodes at a given depth". I only understood the intent of this iterator after seeing Sergey's comment about the constness of the depth parameter. I would even suggest renaming the iterator to something like `OctreeFixedDepthIterator` but I leave this suggestion to your scrutiny regarding it's adoption or not.

Reply:
I update the description.

I keep the class name as is for the moment because I feel that it will clutter the review, but I keep it in mind as a TODO.